### DISCUSSION TITLE
pods_pick_select2_overrides does not work with callback options

### DISCUSSION THREAD

**Discussion by bkonia** [pods-framework/pods#6768] [Ideas]
The pods_pick_select2_overrides filter works well for overriding simple options. However, it doesn't work with the Select2 callback options, because it passes the function name as a string, but Select2 is expecting it to be passed as a function (without quotes).

**Comment by pglewis** [comment]
I'll need to look into this but I'm not sure if there is an elegant solution.  We are marshalling data between PHP on the server side to Javascript on the client side.  JSON does not allow functions, it is a data-interchange format only, and there is no way to represent Javascript functions on the PHP side except as a string regardless.

**Comment by pglewis** [comment]
It might help to know your use case, there may be another direction to attack from.

**Comment by bkonia** [comment]
I think you just need to specify the function name without quotes. In the Select2 docs, they show examples of this for the callback functions:

```
$(".js-example-templating").select2({
  templateResult: formatState
});
```

I have a few different use cases. For example, I was trying to use the matcher callback to customize the search results list. In another example, I was trying to use the templateResut and templateSelection callbacks to customize the appearance of the data, adding icons, etc...

**Comment by pglewis** [comment]
The problem with that is two-fold: 

1) There is no facility on the PHP side to set the value portion of a key/value pair to a bare function name 

2) We're using JSON to send the data, which restricts the value portion to string, number, object, array, true, false, or null (http://www.json.org/).  

Since you have custom Javascript for your callback functions it makes more sense to me if we expose a convenient way to set the callbacks on the Javascript side.

**Comment by bkonia** [comment]
Yes, that would be a fine solution.
